proposals/devcontainer-features-distribution.md

```diff
@@ -0,0 +1,131 @@
+# Community Contribution and Discovery of Dev Container Features
+
+This specification defines a pattern where community members and organizations can author and self-publish [dev container 'features'](./devcontainer-features.md). 
+
+Goals include:
+
+- For community authors, a "self-served" mechanism for dev container feature publishing.
+- For community members, a largely self-served mechanism for marking features as "discoverable" by [supporting tools](../docs/specs/supporting-tools.md).
+- For users, ease of discoverability of features via [supporting tools](../docs/specs/supporting-tools.md).
+- For users, the abilty to validate the origin of the asset and its integrity when compared to previous pulls. 
+- For users, the ability for a user to pin to a particular version (absolute, or semantic version) of a feature to allow consistent, repeatable environments.
+
+## Source Code
+
+Features source code is stored in a git repository.
+
+For ease of authorship and maintenance, [1..n] features can share a single git repository.  This set of features is referred to as a collection, and will share the same [`devcontainer-collection.json`](#devcontainer-collection.json) file and 'namespace' (eg. `owner/repo`).
+
+Source code for the set follows the example file structure below:
+
+```
+.
+├── README.md
+├── src
+│   ├── dotnet
+│   │   ├── devcontainer-feature.json
+│   │   ├── install.sh
+│   │   └── ...
+|   ├
+│   ├── go
+│   │   ├── devcontainer-feature.json
+│   │   └── install.sh
+|   ├── ...
+│   │   ├── devcontainer-feature.json
+│   │   └── install.sh
+├── test
+│   ├── dotnet
+│   │   ├── devcontainer-feature.json
+│   │   └── ...
+│   └── go
+│   |   └── test.sh
+|   ├── ...
+│   │   └── test.sh
+├── ...
+```
+
+Where `src` is a directory containing a sub-folder with the name of the feature (e.g. `dotnet` or `go`) with at least a file named `devcontainer-feature.json` that contains the feature metadata. Each feature sub-directory also contains an `install.sh` script that implementing tools will use to install the feature.  Each sub-directory should be named such that it matches the `id` field of the `devcontainer-feature.json`.  Other files can also be included in the sub-directory, and will be packaged along side the two required files.
```

Comment:
If a feature (say `src/dotnet`) is trying to execute a file (`src/helper.sh`) which is not inside the feature's directory (`src/dotnet`), then I believe that's not (currently) acceptable and won't work?  

The highlighted line implicitly points that, but do we want to explicitly mention?